src/object/srv_obj.c

```diff
@@ -2425,8 +2425,8 @@ ds_obj_ec_rep_handler(crt_rpc_t *rpc)
 	dkey = (daos_key_t *)&oer->er_dkey;
 	iod = (daos_iod_t *)&oer->er_iod;
 	iod_csums = oer->er_iod_csums.ca_arrays;
-	rc = vos_update_begin(ioc.ioc_coc->sc_hdl, oer->er_oid, oer->er_epoch_range.epr_hi, 0,
-			      dkey, 1, iod, iod_csums, 0, &ioh, NULL);
+	rc = vos_update_begin(ioc.ioc_coc->sc_hdl, oer->er_oid, oer->er_epoch_range.epr_hi,
+			      VOS_OF_REBUILD, dkey, 1, iod, iod_csums, 0, &ioh, NULL);
```

Comment:
Maybe this is ok since we are writing a full stripe.  But is there any potential for partial stripe conflict with subsequent rebuild if we do this?   There is one additional bit we could use in VOS to put these in yet another range if it would help.

Reply:
To be clear, this would mean a follow-on to this patch with a range for EC aggregation that sits below a range for rebuild logically.

https://github.com/daos-stack/daos/blob/9bcb9aa32ed089f993885951acdef1b09762c693/src/vos/evt_priv.h#L12-L25

so basically it would use 10 bits

Reply:
and then one just would need to change evt_rect_read and evt_rect_write

Reply:
the critical log in the test we hit is -
06/03-03:57:00.61 boro-35 DAOS[1279732/10/160374] vos  ERR  src/vos/evtree.c:2715 evt_ent_array_fill() Same epoch partial overwrite not supported: 8000000000030000-800000000003ffff@17f6f40711100000.16383-INF overlaps with 8000000000000000-800000000003ffff@17f6f40711100000.16383-INF
。。。
06/03-03:57:00.63 boro-34 DAOS[1271734/12/263756] object ERR  src/object/srv_ec_aggregate.c:1411 agg_peer_update_ult() 946600381037674498.0.26.1 pidx 0 to peer 1, ea_status DER_VOS_PARTIAL_UPDATE(-2045): 'Same epoch partial overwrite of VOS array value disallowed'

It possibly be happened in the case with EC aggregate leader switch during reint/rebuild. 
So I added the VOS_OF_REBUILD flag to the EC agg peer update RPC handler's vos update, with that change no such err observed any more in following test.

Reply:
So my concern stems from prior conversations with @wangdi1 and the reason for the original overwrite patch.   If there can be any conflict between what we write here and maybe a subsequent partial stripe and a rebuild, they would be at the same minor epoch.   Anyway, not sure it's a potential issue but the above would ensure we avoid it.  I think we can land this and think about it afterwards.